src/language_features/code_lens.rs

```diff
@@ -32,10 +32,54 @@ pub fn text_document_code_lens(meta: EditorMeta, ctx: &mut Context) {
         partial_result_params: PartialResultParams::default(),
     };
     ctx.call::<CodeLensRequest, _>(meta, req_params, |ctx: &mut Context, meta, result| {
-        editor_code_lens(meta, result, ctx)
+        editor_code_lens(meta.clone(), result.clone(), ctx);
+        inlay_hints_from_editor_code_lens(meta, result.unwrap_or_default(), ctx)
     });
 }
 
+fn inlay_hints_from_editor_code_lens(
+    meta: EditorMeta,
+    mut lenses: Vec<CodeLens>,
+    ctx: &mut Context,
+) {
+    let document = match ctx.documents.get(&meta.buffile) {
+        Some(document) => document,
+        None => {
+            return;
+        }
+    };
+    lenses.sort_by_key(|lens| lens.range.start);
+
+    let inlay_hints_from_lenses = lenses
+        .iter()
+        .map(|lens| {
+            let label = lens.command.as_ref().map_or("", |v| &v.title);
+            if label.trim().len() > 0 {
+                let position =
+                    lsp_position_to_kakoune(&lens.range.start, &document.text, ctx.offset_encoding);
+                let line = position.line;
+                let col = position.column;
+                editor_quote(&format!(
+                    "{line}.{col}+0|{{InlayHint}}[\u{1F50E} {}] {{InlayHint}}",
```

Comment:
You escape special tuple-characters (`|` and backslash) in `label` but we also need to escape `{`,
because it's interpreted as markup string.

There are two ways to do that (we still use both but should converge eventually).

`:doc faces` says

> The {\} string disables markup processing for the rest of the line, and can be used to avoid having to escape text that might be mistaken for markup instructions.

So we could end the string with `{{\\}} {}]`,

or add a `.replace('{', "\\{")`. This is what the inlay-hints implementation does.
Unfortunately we can't use `escape_kakoune_markup` because that would escape backslashes too much. That's a limitation in Kakoune, I'll raise an issue later.